app/controllers/application_controller.rb

```diff
@@ -2,4 +2,21 @@ class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+  helper_method :current_user, :logged_in?
+
+  def current_user
+    @user = User.find_by(id: session[:user_id]) if session[:user_id]
+  end
+
+  def logged_in?
+    !!session[:user_id]
+  end
```

Comment:
Rather than `!!session[:user_id]` I'd do `!!current_user` to ensure that the user_id in the session is valid (i.e. matches a user id in the database.)